### PR TITLE
All redirect operations should have content-type as json

### PR DESCRIPTION
**Pull request by gklp**
related with https://github.com/goharbor/harbor/issues/8197.

Every redirection request behind of apis has this problem. This pr can be general solution.

### CODE REVIEW

**Comment by gklp** [comment]
Hello @wy65701436 , @reasonerjt  , cloud you review this pr or is there anyone for that :)

File: src/common/api/base.go [hunk 1]
```diff
@@ -112,6 +112,7 @@ func (b *BaseAPI) Redirect(statusCode int, resouceID string) {
 	requestURI := b.Ctx.Request.RequestURI
 	resourceURI := requestURI + "/" + resouceID
 
+	b.Ctx.ResponseWriter.Header().Set("Content-Type", "application/json")
```
Comment (wy65701436) [inline src/common/api/base.go]: I'd like to modify the robot api only, not for the base method.

Comment (gklp) [reply src/common/api/base.go]: Okey i will change it as you mentioned.

Comment (wy65701436) [reply src/common/api/base.go]: thanks, as we're closing to release, is it possible to finish it by the end next Wed?

Comment (gklp) [reply src/common/api/base.go]: yes i will do it.

Comment (gklp) [reply src/common/api/base.go]: Done. @wy65701436

**Comment by wy65701436** [comment]
@gklp could you please fix the DCO?

**Comment by gklp** [comment]
> @gklp could you please fix the DCO?

Done @wy65701436 .

**Pull request merged**